### ISSUE TITLE
little bug in tensor_regression_layer_pytorch.ipynb

### ISSUE THREAD

**Issue by segalinc** [JeanKossaifi/tensorly-notebooks#13]
Hi Jean, 

just wanted to point out the in the example tensor_regression_layer_pytorch.ipynb
in the TRL layer forward pass this line `regression_weights = tl.tucker_to_tensor(self.core, self.factors)`
should instead be `regression_weights = tl.tucker_to_tensor((self.core, self.factors))` otherwise you get an error. 
This happened to me while working on my code using latest version 

Let me know if you get it too running the example

**Comment by JeanKossaifi** [comment]
Hi Christina,

Thanks for reporting, you are completely right! The issue should now have been fixed in 6af351c30928f07eab740a460272f1f2ac6dd5df.

As a side note, we now provide well tested PyTorch Tensor Regression Layers in [TensorLy-Torch](http://tensorly.org/torch/dev/index.html):
* API: http://tensorly.org/torch/dev/modules/api.html#tensor-regression-layers
* Example: http://tensorly.org/torch/dev/user_guide/trl.html

These also support tensor hooks such as tensor dropout or rank regularization (lasso).

**Comment by segalinc** [comment]
Awesome, trying it now ,thanks!

Would you recommend to use rank='same'  or idk (512,3,3,output_cls) for the
rank for networks like VGG or resnet18?

**Comment by JeanKossaifi** [comment]
Based on the tuple you show for rank, I'm assuming this is for Tucker. I would say `rank='same'` is a good start. Alternatively, `rank=kernel.shape` should always work well.
Ideally you want to reduce the rank to benefit from the low-rank regularization (e.g. `rank=0.75`).

Depending on the problem, with fine-tuning, you should be able to reach `rank=0.5` without loss of performance (the low-rank regularization may even help generalize better with longer fine-tuning).
Let me know of any feedback you may have! :)
